server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -233,7 +233,12 @@ public static int geoHashLevelsForPrecision(String distance) {
      * @return The normalized longitude.
      */
     public static double normalizeLon(double lon) {
-        return centeredModulus(lon, 360);
+        if (lon > 180d || lon <= -180d) {
+            return centeredModulus(lon, 360);
+        } else if (lon == -0d) {
```

Comment:
hmm this looks a bit weird given that `-0 == +0` evaluates to true

Reply:
if you want to avoid negative zero as a return value, then you could do something like `else { return lon + 0.0; }`